main.js

```diff
@@ -44,127 +44,134 @@ it is available then var.The variable is available inside a block and it's child
 You can define the same variable name inside a child without produce a conflict with the variable of the parent block.*/
 let variable = 1234;
 
-// create adapter instance wich will be used for communication with controller
-let adapter;
-function startAdapter(options) {
-	options = options || {};
-	Object.assign(options, {
-        // name has to be set and has to be equal to adapters folder name and main file name excluding extension
-        name: adapterName,
-        // is called when adapter shuts down - callback has to be called under any circumstances!
-        unload: function (callback) {
-            try {
-                adapter.log.info('cleaned everything up...');
-                callback();
-            } catch (e) {
-                callback();
+// define adapter class wich will be used for communication with controller
+class MyAdapter extends utils.Adapter {
+    constructor(options) {
+        let adapterOptions = {
+            // name has to be set and has to be equal to adapters folder name and main file name excluding extension
+            name: adapterName,
+
+            // is called when adapter shuts down - callback has to be called under any circumstances!
+            unload: this._unload,
+
+            // is called if a subscribed object changes
+            objectChange: this._objectChange,
+
+            // is called if a subscribed state changes
+            stateChange: this._stateChange,
+
+            // Some message was sent to adapter instance over message box. Used by email, pushover, text2speech, ...
+            message: this._message,
+
+            // is called when databases are connected and adapter received configuration.
+            // start here!
+            ready: this._main
+        }
+
+        super(Object.assign(adapterOptions, options || {}));
+    }
+
+    _unload(callback) {
```

Comment:
The leading underscore looks ugly, I'd remove them

Reply:
i like them ;-)

Reply:
Okay then :D Which meaning are you trying to convey? Something like "this is a private method"?

Reply:
yes